Educado-App/educado-mobile
# Dev to stage [IMPORTANT]

## Description

👏 Finally, I am able to present to you, a dev branch that should resemble staging, and a few added features such as linting, toast notifications, updated endpoints, refactors etc.

**THIS IS VERY IMPORTANT THAT YOU ALL CHECK YOUR FILES, AND THAT THE CODE IS HOW YOU WANT IT TO BE, AND THAT THE APP BEHAVES AS INTENDED. THIS PR IS CRUCIAL TO ELIMINATE CONFLICTS BETWEEN DEV AND STAGING.**

### AND LETS AGREE TO <ins>NEVER</ins> PUSH DIRECTLY TO STAGING AGAIN, IF THE CHANGES ARE NOT IN DEV!!!

## Changes

- made urls in userApi.js and api.js identical to staging
- linting added to config files
- linting added to tests
- workflow updated in yml files
- eslintr updated to single quotes and semicolons
- singleIcon.svg removed
- singleIcon.png moved to images¨
- removed comments and stack.screen error screen in App.js **(Please check this is correct)**
- changed routes and return data in api.js  **(Please check this is correct)**
- changed routes and return data in userApi.js  **(Please check this is correct)**
- build changes added in app.js  **(Please check this is correct)**
- moved mp3 files to sound folder
- removed unused testvideo.mp4
- changed imports of video in content.js
- replaced hardcoded colors, as done in staging progressbar.js and customrating.js
- time added to explorecard.js
- added alert notification to base screen
- added dialogNotification.js
- added alert notification to educadoModal.js
- refactored import of icon in IconHeader.js
- Added toastNotification.js
- added format hours to coursecard.jsx
- dialogNotifications added to enterNewPasswordScreen
- completedCourses set in asyncstorage on LoginForm
- DialogNotification and alertNotification added to RegisterForm
- updated user info setting in registerForm
- Toast notification added to ResetPassword
- updated fetching on changeEmailButton, changeFirstNameButton, and changeLastNameButton
- refactor of deleteAccountButton
- Replaced icon with SVG in returnButton
- updated navigation in sectionCard
- welcomeSlider replaced icon with svg
- updated jestSetup to remove duplicate mocks from test files
- Changes in package.json  **(Please check this is correct)**
- toast notification added to coursescreen
- update on exerciseScreen and replaced hardcoded colors with tailwind colors
- removed keys from TextImageLecturerScreen  **(Please check this is correct)**
- updated fetching on Profile
- Welcome.js changed navigation **(Please check this is correct)**
- exercise completion functions added to UtilityFunctions
- updated endpoints in api.test.js
- updated snaps
- added jest.useRealTimers to mock setTimeout on LoginForm
- updated fields in exerciseScreenMock
- added mocks to CourseScreen test
- SectionScreen mocks added
- FormatHours function added to UtilityFunctions.test.js
- added tailwindcss to package.json, to fix error as pr. stackoverflow

## Related Issues

None.

## Checklist

- [X] Code has been tested locally and passes all relevant tests.
- [X] Documentation has been updated to reflect the changes, if applicable.
- [X] Code follows the established coding style and guidelines of the project.
- [X] All new and existing tests related to the changes have passed.
- [X] Any necessary dependencies or new packages have been properly documented.
- [X] Pull request title and description are clear and descriptive.
- [X] Reviewers have been assigned to the pull request.
- [X] Any potential security implications have been considered and addressed.
- [X] Performance impact of the changes has been evaluated, if relevant.

## Screenshots (if applicable)

None.

## If mobile/frontend pull request, what version of the backend is it stable, and running on? 

Branch: dev

Commit id: 1fd784b2965a2ea0fef1b0a270190e98f01c9a85

## Notes for Reviewers

**IT IS VERY IMPORTANT YOU CHECK YOUR OWN FILES, AND SEE THAT THEY ARE HOW THEY SHOULD BE. PLEASE ALSO CHECK YOUR FUNCTIONALITY BEHAVES AS INTENDED**

**SOME OF THE FUNCTIONALITIES FOR EXERCISE COMPLETION IS NOT YET ADAPTED TO THE NEW USER MODEL, BUT THIS WILL BE DONE THIS WEEK**

**SOME IMPORTS HAS BEEN CHANGED TO REQUIRE DUE TO LINTER ISSUES. THIS CAN BE FIXED AT A LATER POINT**
